### PR TITLE
Rs/sparse integration test clean 2

### PR DESCRIPTION
**Pull request by robertgshaw2-neuralmagic**
SUMMARY:
Adds support for end-to-end sparsity integration tests for 2:4 and unstructured using the logits comparison method

TEST PLAN:
- Correctness testing by comparing "running dense" vs "running magic_wand" kernels
- Memory usage by comparing the resulting size of the kv cache, and checking that the vllm cache is bigger (and thus the weights are smaller)

### CODE REVIEW

**Review by tlrmchlsmth — APPROVED** [review]
LGTM when green. Could you fill out the PR summary as well before landing?

File: tests/models/test_compressed_memory.py
Comment (robertgshaw2-neuralmagic) [reply tests/models/test_compressed_memory.py:23]: going to address in other PR

File: tests/models/test_compressed.py
Comment (robertgshaw2-neuralmagic) [reply tests/models/test_compressed.py:24]: going to address in other PR

**Review by andy-neuma — APPROVED** [review]
checked `testmo` run and the "test compressed" ran ... thanks!

**Review by mgoin — APPROVED** [review]

**Pull request merged**
